skimage/restoration/_cycle_spin.py

```diff
@@ -0,0 +1,181 @@
+from multiprocessing import cpu_count
+from itertools import product
+
+import numpy as np
+from .._shared.utils import warn
+
+has_futures = True
+try:
+    from concurrent import futures
+except ImportError:
+    try:
+        import futures
+    except ImportError:
+        has_futures = False
```

Comment:
I would suggest using dask, since we already depend on it and it already does all this wrangling for us.